### PR TITLE
Chore/column editor show types from extensions schema

### PR DESCRIPTION
**Pull request by joshenlim**
Show postgres enumerated types from the `extensions` schema on the table and column side panel editor (for the Table Editor and Database pages)

Database extensions that are installed through the dashboard on the database/extensions page are, most of the time, installed by default in the extensions schema (it's the default dropdown option) unless the extension has a schema that it's required to be in, or the user changes it to be installed in another schema. 

If the installed extension (e.g vector) has enumerated types, the Table Editor then can access those types for users to assign them to columns, without having the user to install them in another schema.

### CODE REVIEW

File: apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/TableEditor.tsx [hunk 1]
```diff
@@ -60,8 +61,13 @@ const TableEditor = ({
...
 
   const realtimeEnabled = useIsFeatureEnabled('realtime:all')
-
-  const enumTypes = meta.types.list((type: PostgresType) => !EXCLUDED_SCHEMAS.includes(type.schema))
+  const { data: types } = usePostgresTypesQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+  const enumTypes = (types ?? []).filter(
+    (type) => !EXCLUDED_SCHEMAS.filter((x) => x !== 'extensions').includes(type.schema)
```
Comment (alaister) [inline apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/TableEditor.tsx:69]: small perf nit: could move the `EXCLUDED_SCHEMAS.filter((x) => x !== 'extensions')` part out of this `.filter()` into its own const (maybe something like `EXCLUDED_SCHEMAS_WITHOUT_EXTENSIONS`) above the component. That way it we can avoid some nested loops :)

File: apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnEditor.tsx [hunk 2]
```diff
@@ -44,6 +44,8 @@ import {
 import ColumnForeignKey from './ColumnForeignKey'
 import ColumnType from './ColumnType'
 import HeaderTitle from './HeaderTitle'
+import { toJS } from 'mobx'
```
Comment (alaister) [inline apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnEditor.tsx:47]: In use?

**Review by alaister — COMMENTED** [review]
🔥

File: apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/TableEditor.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/TableEditor.tsx:69]: gotcha!

File: apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnEditor.tsx (hunk 2 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnEditor.tsx]: nope! good catch 🙏

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**